src/services/transfer/transfer-out-ehr-core.js

```diff
@@ -68,39 +57,41 @@ export async function transferOutEhrCore({
       'EHR has been successfully sent'
     );
   } catch (error) {
-    if (error instanceof EhrUrlNotFoundError) {
-      await updateConversationStatus(conversationId, Status.MISSING_FROM_REPO);
-    }
-    if (error instanceof DownloadError) {
-      await updateConversationStatus(conversationId, Status.EHR_DOWNLOAD_FAILED);
-    }
-    logError('EHR transfer out request failed', error);
+    await handleCoreTransferError(error, conversationId);
   }
 }
 
 const getEhrCoreAndUpdateMessageIds = async (nhsNumber, conversationId) => {
-  const { ehrCore, fragmentMessageIds } = await getEhrCoreAndFragmentIdsFromRepo(
-    nhsNumber,
-    conversationId
-  );
-
-  let { ehrCoreWithUpdatedMessageId, newMessageId } = await updateMessageIdForEhrCore(ehrCore);
-  logInfo(`Replaced message id for ehrCore`);
-
-  if (fragmentMessageIds?.length > 0) {
-    await createNewMessageIdsForAllFragments(fragmentMessageIds);
-    logInfo(`Created new message id for all fragments`);
-    ehrCoreWithUpdatedMessageId = await updateReferencedFragmentIds(ehrCoreWithUpdatedMessageId);
-    logInfo(`Replaced fragment id references in ehrCore`);
-  }
+  const { ehrCore, fragmentMessageIds } = await getEhrCoreAndFragmentIdsFromRepo(nhsNumber, conversationId);
+  const ehrCoreMessageId = await parseMessageId(ehrCore);
+  const messageIdReplacements = await createNewMessageIds([ehrCoreMessageId, ...fragmentMessageIds]);
+  const ehrCoreWithUpdatedMessageId = replaceMessageIdsInObject(ehrCore, messageIdReplacements);
+  const newMessageId = getNewMessageIdForOldMessageId(ehrCoreMessageId, messageIdReplacements);
+
   return { ehrCoreWithUpdatedMessageId, newMessageId };
 };
 
 const isEhrRequestDuplicate = async conversationId => {
-  const previousTransferOut = await getRegistrationRequestStatusByConversationId(conversationId);
+  const previousTransferOut = await getRegistrationRequestByConversationId(conversationId);
+
   if (previousTransferOut !== null) {
-    logInfo(`EHR out transfer with conversation ID ${conversationId} is already in progress`);
+    logWarning(`EHR out transfer with conversation ID ${conversationId} is already in progress`);
     return true;
   }
+
   return false;
 };
+
+const handleCoreTransferError = async (error, conversationId) => {
+  switch (true) {
```

Comment:
First time I have seen `switch (true)` used before.
(O.o)

Reply:
As `error instanceof XYZError` evaluates to either true or false, we want to switch on true for the case to be run!